Solutions/Marain.TenantManagement.Abstractions/Marain/TenantManagement/Internal/TenantManagementService.cs

```diff
@@ -444,6 +445,57 @@ public async Task<ServiceManifestRequiredConfigurationEntry[]> GetServiceEnrollm
             return requirements.ToArray();
         }
 
+        /// <inheritdoc/>
+        public async Task AddConfigurationAsync(ITenant tenant, ConfigurationItem[] configurationItems)
+        {
+            if (tenant == null)
+            {
+                throw new ArgumentNullException(nameof(tenant));
+            }
+
+            tenant.EnsureTenantIsOfType(MarainTenantType.Client, MarainTenantType.Delegated, MarainTenantType.Service);
```

Comment:
Why bother? This is basically saying you can attach arbitrary config to any type of tenant.

However, at the moment I can't see why we'd be attaching config to Delegated or Service tenants. I don't think we should ever be directly manipulating delegated tenants, as these are created automatically as part of enrollment on behalf of an existing client tenant. And at the moment, we don't have any scenarios where service tenants are actually used to store configuration. So maybe this should be limited to client tenants only?

Reply:
Limited to client tenant